lib/shopify-cli/method_object.rb

```diff
@@ -64,10 +64,27 @@ module ClassMethods
       # initializer or to `call`. If the keyword argument matches the name of
       # property, it is forwarded to the initializer, otherwise to call.
       #
-      def call(*args, **kwargs)
-        properties.keys.yield_self do |properties|
-          new(**kwargs.slice(*properties))
-            .call(*args, **kwargs.slice(*(kwargs.keys - properties)))
+      ruby2_keywords def call(*args)
+        # This is an extremely complicated case of delegation. The method wants
+        # to delegate arguments, but to have control over which keyword
+        # arguments are delegated. I'm not sure the forward and backward
+        # compatibility of this unusual form of delegation has really been
+        # explored or there's any good way to support it. So I have done
+        # done something hacky here and I'm looking at the last argument and
+        # modifying the package of arguments to be delegated in-place.
+        if args.last.is_a?(Hash)
```

Comment:
A way to make sure it's a keyword Hash and not a positional Hash as last argument is `args.last.is_a?(Hash) && Hash.ruby2_keywords_hash?(args.last)`. `Hash.ruby2_keywords_hash?` is also provided by the `ruby2_keywords` gem.

Reply:
Isn't `Hash.ruby2_keywords_hash? ` always `false`?

Reply:
Mmh, but that wouldn't work on Ruby < 2.7, so probably best to keep as-is.

Reply:
Context is this gem supports back to 2.5, as it's a tool that may be used by a broad-range of end-users.